src/apps/loan-list/materials/selectable-material/selectable-material.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { FC, useCallback, MouseEvent } from "react";
+import { formatDate, isDigital } from "../../utils/helpers";
+import { useText } from "../../../../core/utils/text";
+import StatusBadge from "../utils/status-badge";
+import { LoanType } from "../../../../core/utils/types/loan-type";
+import { FaustId } from "../../../../core/utils/types/ids";
+import fetchMaterial, { MaterialProps } from "../utils/material-fetch-hoc";
+import fetchDigitalMaterial from "../utils/digital-material-fetch-hoc";
+import CheckBox from "../../../../components/checkbox/Checkbox";
+import { useModalButtonHandler } from "../../../../core/utils/modal";
+import MaterialDetailsModal from "../../modal/material-details-modal";
+import MaterialDetails from "../../modal/material-details";
+import StatusMessage from "./StatusMessage";
+
+interface SelectableMaterialProps {
+  loan: LoanType;
+  disabled?: boolean;
+  materialsToRenew: FaustId[];
+  onChecked?: (faust: FaustId) => void;
+}
+
+const SelectableMaterial: FC<SelectableMaterialProps & MaterialProps> = ({
+  loan,
+  material,
+  disabled,
+  onChecked,
+  materialsToRenew
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const { dueDate, faust, identifier } = loan;
+  const { authors, materialType, year, title } = material || {};
+
+  const selectListMaterial = useCallback(
+    (e: MouseEvent) => {
+      e.stopPropagation();
+      open(faust || identifier || "");
+    },
+    [faust, identifier, open]
+  );
+
+  return (
+    <>
+      <li>
+        <div
+          className={`list-materials ${
+            disabled ? "list-materials--disabled" : ""
+          }`}
+        >
+          <div className="mr-32">
+            {faust && onChecked && (
+              <CheckBox
+                onChecked={() => onChecked(faust)}
+                id={faust}
+                selected={materialsToRenew?.indexOf(faust) > -1}
```

Comment:
What happens here if materialsToRenew is undefined? indexOf will be called on undefined?

Reply:
It returns undefined, should i wrap it in a Boolean(...